utils.sh

```diff
@@ -380,18 +380,43 @@ EOF
 function verify_pull_secret() {
   # Do some PULL_SECRET sanity checking
   if [[ "${OPENSHIFT_RELEASE_IMAGE}" == *"registry.svc.ci.openshift.org"* ]]; then
-      if [[ "${PULL_SECRET}" != *"registry.svc.ci.openshift.org"* ]]; then
-          echo "Please get a valid pull secret for registry.svc.ci.openshift.org."
+      if [[ ${#CI_TOKEN} = 0 ]]; then
+          error "Please login to https://api.ci.openshift.org and copy the token from the login command from the menu in the top right corner to set CI_TOKEN."
           exit 1
       fi
   fi
 
-  if [[ "${PULL_SECRET}" != *"cloud.openshift.com"* ]]; then
-      echo "Please get a valid pull secret for cloud.openshift.com."
+  if ! grep -q cloud.openshift.com "${PERSONAL_PULL_SECRET}"; then
+      error "No cloud.openshift.com pull secret in ${PERSONAL_PULL_SECRET}"
+      error "Get a valid pull secret (json string) from https://cloud.redhat.com/openshift/install/pull-secret"
       exit 1
   fi
 }
 
+function write_pull_secret() {
+    if [ ${#PULL_SECRET} != 0 ]; then
+        echo "Using PULL_SECRET variable. Consider switching to PERSONAL_PULL_SECRET and CI_TOKEN."
+        set +x
+        echo "${PULL_SECRET}" > ${PULL_SECRET_FILE}
+        set -x
+        return
```

Comment:
I tested without making changes to my config, e.g with the existing PULL_SECRET and the mirroring fails like:

```
error: unable to upload blob sha256:a03401a44180b6581a149376d6fd2d5bd85d938445fd5b5ad270e14ddde4937c to virthost.ostest.test.metalkube.org:5000/localimages/local-release-image: Post "https://virthost.ostest.test.metalkube.org:5000/v2/localimages/local-release-image/blobs/uploads/": no basic auth credentials
error: unable to upload blob sha256:97c7aeeb52fe8b1a310c9b1ec54763d1affbba48be9298b6476f5bc13896fbf5 to virthost.ostest.test.metalkube.org:5000/localimages/local-release-image: Post "https://virthost.ostest.test.metalkube.org:5000/v2/localimages/local-release-image/blobs/uploads/": no basic auth credentials
info: Mirroring completed in 56.2s (0B/s)
error: one or more errors occurred while uploading images
```

Which I guess is expected, since we return here before doing the oc registry login etc.

I wonder if we should consider just removing the old interface, and forcing everyone to switch to the new interface?  Then at least everyone can use the same path, and we can ensure it stays working in CI?

Reply:
I thought about that, but wasn't sure how CI was being configured so I wasn't sure what to do to update it. Are the scripts to get the pull secret for CI in this repo somewhere?

Reply:
No, it's part of the of the multistage test here: https://github.com/openshift/release/blob/d741522e927fb687a1357ebfdac7593e9fd353ef/ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh#L46

Reply:
OK, so in order for this PR to pass CI at all we need it to continue to support PULL_SECRET. We could update the job definition separately, then come back and remove the PULL_SECRET support here.

Reply:
I've updated this section to always merge the $PULL_SECRET with the local registry credentials instead of assuming that PULL_SECRET is complete on its own.